File: afc_saaspro_tax/afc_rest_apis/javascript/scripts/model/CalculateWithOverridesRequest.js
/**
 * Parameter for the tax calculation with override APIs.
 */

function CalculateWithOverridesRequest() {

    this.Transaction = null;
    this.Overrides = [];

    /**
     * Sets the transaction data.
     * @param transaction Transaction object.
     * @return Reference to this object for method chaining. 
     */
    this.transaction = function (transaction) {
        this.Transaction = transaction;
        return this;
    }

    /**
     * Transaction data.
     * @return Transaction object.
     */
    this.getTransaction = function () {
        return this.Transaction;
    }

    /* Reference to this object for method chaining.*
     * Sets the transaction data.
     * @param transaction Transaction object. 
     */
    this.setTransaction = function (transaction) {
        this.Transaction = transaction;
    }

    /**
     * Sets the list of tax rate overrides to be applied in tax calculations.
     * @param overrides Array of TaxRateOverrideInfo
     * @return Reference to this object for method chaining. 
     */
    this.overrides = function (overrides) {
        this.Overrides = overrides;
        return this;
    }

    /**
     * Adds an tax rate override to this request.
     * @param overridesItem TaxRateOverrideInfo object.
     * @return Reference to this object for method chaining. 
     */
    this.addOverride = function (overridesItem) {
        this.Overrides.push(overridesItem);
        return this;
    }

    /**
     * List of tax rate overrides to be applied in tax calculations.
     * @return Array of TaxRateOverrideInfo
     */
    this.getOverrides = function () {
        return this.Overrides;
    }

    /* Reference to this object for method chaining.*
     * Sets the list of tax rate overrides to be applied in tax calculations.
     * @param overrides Array of TaxRateOverrideInfo 
     */
    this.setOverrides = function (overrides) {
        this.Overrides = overrides;
    }
}

File: afc_saaspro_tax/afc_rest_apis/java/src/avalara/comms/rest/sample/model/CalculateWithOverridesRequest.java
package avalara.comms.rest.sample.model;

import com.google.gson.annotations.SerializedName;
import java.util.ArrayList;
import java.util.List;

/**
 * Parameter for the tax calculation with override APIs.
 */
public class CalculateWithOverridesRequest {

    @SerializedName("Transaction")
    private Transaction transaction = null;

    @SerializedName("Overrides")
    private List<TaxRateOverrideInfo> overrides = new ArrayList<>();

    /**
     * Sets the transaction data.
     * @param transaction
     * @return 
     */
    public CalculateWithOverridesRequest transaction(Transaction transaction) {
        this.transaction = transaction;
        return this;
    }

    /**
     * Transaction data.
     * @return transaction
     */
    public Transaction getTransaction() {
        return transaction;
    }

    /**
     * Sets the transaction data.
     * @param transaction 
     */
    public void setTransaction(Transaction transaction) {
        this.transaction = transaction;
    }

    /**
     * Sets the list of tax rate overrides to be applied in tax calculations.
     * @param overrides
     * @return 
     */
    public CalculateWithOverridesRequest overrides(List<TaxRateOverrideInfo> overrides) {
        this.overrides = overrides;
        return this;
    }

    /**
     * Adds an tax rate override to this request.
     * @param overridesItem
     * @return 
     */
    public CalculateWithOverridesRequest addOverride(TaxRateOverrideInfo overridesItem) {
        this.overrides.add(overridesItem);
        return this;
    }

    /**
     * List of tax rate overrides to be applied in tax calculations.
     * @return overrides
     */
    public List<TaxRateOverrideInfo> getOverrides() {
        return overrides;
    }

    /**
     * Sets the list of tax rate overrides to be applied in tax calculations.
     * @param overrides 
     */
    public void setOverrides(List<TaxRateOverrideInfo> overrides) {
        this.overrides = overrides;
    }
}
